src/Ens.ts

```diff
@@ -11,18 +10,61 @@ import {
 import Contract from './utils/contract';
 import contentHash from 'content-hash';
 import EnsNetworkMap from 'ethereum-ens-network-map';
-import { ResolutionResponse, CryptoRecords, SourceDefinition } from './publicTypes';
-import { isNullAddress } from './utils';
+import { Provider, EnsSource, EnsConfig } from './types/publicTypes';
+import { constructRecords, ensureConfigured, isNullAddress } from './utils';
+import NamingService from './interfaces/NamingService';
+import FetchProvider from './FetchProvider';
+import Namehash from './utils/Namehash';
+
+export default class Ens implements NamingService {
+  static readonly UrlMap: BlockhanNetworkUrlMap = {
+    1: 'https://mainnet.infura.io/v3/d423cf2499584d7fbe171e33b42cfbee',
+    3: 'https://ropsten.infura.io/v3/d423cf2499584d7fbe171e33b42cfbee'
+  };
+
+  static readonly NetworkNameMap = {
+    mainnet: 1,
+    ropsten: 3,
+    rinkeby: 4,
+    goerli: 5,
+  };
 
-export default class Ens extends EthereumNamingService {
   readonly name = NamingServiceName.ENS;
+  readonly network: number;
+  readonly url: string | undefined;
```

Comment:
I don't think we need to have this field in our naming services. We just pass it to the provider constructor and don't use it anywhere else.